cmd/iam.go

```diff
@@ -532,6 +532,17 @@ func (sys *IAMSys) DeletePolicy(ctx context.Context, policyName string, notifyPe
 		return errServerNotInitialized
 	}
 
+	m := map[string]PolicyDoc{}
+	if err := sys.store.loadPolicyDocs(ctx, m); err != nil {
+		return err
+	}
```

Comment:
I am not sure what this code is needed, is it because it is possible to upload a custom policy with an existing name ?

Reply:
if policies are persisted, deletion should be allowed, so below `for` loop checks if inbuilt policies are not persisted and we are trying to delete, don't allow. Not sure if its confusing.

Reply:
@vadmeste are you saying this check itself is wrong?

Reply:
> @vadmeste are you saying this check itself is wrong?

I just don't see why it is needed

Reply:
I understand `loadPolicyDocs` loads persisted policies in the map passed i.e. `m`.
https://github.com/minio/minio/blob/master/cmd/iam-etcd-store.go#L181
https://github.com/minio/minio/blob/master/cmd/iam-object-store.go#L170

Reply:
oh wait this can make the entire DeletePolicy() slow. It's better to validate if the default policies persist on disk individually?.

Well, you don't even have to read you could simply "checkIAMConfig" by doing a GetObjectInfo(). 

Instead, reloading all policies into a local map?

Reply:
In fact, I still don't understand why sys.store.loadPolicyDocs() is needed. 

My guess is that, currently, users are able to create a custom policy with a 'readwrite' name and the name is one of the inbuilt policy, @shtripat can you confirm if that is the case ?

Reply:
Actually you are loading entire set of policies which is unexpected. Let me send a PR here to your branch.

Reply:
Does https://github.com/minio/minio/pull/17264#discussion_r1205293504 make sense?